.lintstagedrc.json

```diff
@@ -0,0 +1,4 @@
+{
+  "*.js": ["eslint --fix", "jest --findRelatedTests", "git add"],
+  "*.{json,md}": ["prettier --write", "git add"]
```

Comment:
- We want ` "*.{js,jsx}"` for ESLint and ` "*.{js,json,md}"` for Prettier, please.
- What is `jest --findRelatedTests` for?
- What does `git add` do? We don't want it to stage automatic formatting at pre-commit hook, just to fail committing so the user can check what Prettier did (with `git diff`) and stage themselves. Thanks!

Reply:
I removed `git add`.

Reply:
> * We want ` "*.{js,jsx}"` for ESLint and ` "*.{js,json,md}"` for Prettier, please.

I agree with linting `.jsx` files using `eslint` but I don't think we need to run `prettier` on `.js,jsx` files since `eslint --fix` will do the same as `prettier --write` 

> * What is `jest --findRelatedTests` for?

It looks up every source and test file that requires your files

Reply:
> `eslint --fix` will do the same as `prettier --write` 

Hmmm. You did mention about [`eslint-plugin-prettier`](https://github.com/prettier/eslint-plugin-prettier) earlier but I didn't quite understand we were doing this until now that you bring it up. Cool! So I guess just add jsx here. Maybe check that other changes I made are consistent with this fact also?

Reply:
> It looks up every source and test file that requires your files

https://jestjs.io/docs/en/cli#findrelatedtests-spaceseparatedlistofsourcefiles reads "Find and run the tests that cover a space..." Are you suggesting we replace the `test` script from package.json (and corresponding step in .circleci/config.yml) with `link-staged`?

Reply:
Wait no. I keep getting confused between pre-commit hook and CI checks. Never mind... I guess this would just add test enforcement for changed files at `git commit`? If so, let's not do this now. It will make committing too slow.

Reply:
> I guess this would just add test enforcement for changed files at `git commit`? If so, let's not do this now. It will make committing too slow.

it will not slow down the commiting cause `jest --findRelatedTests` will run only the tests on the changed staged files not the whole test files, also it makes sense to test before commiting to ship the code with confidence and avoiding catching failed tests in CI instead you can catch it locally before

Reply:
It's already kind of slow. Running Jest will definitely add more wait time for commits, which may push people to start skipping the hook. Tests will be run at CI anyway so there's no need for this now and it's beyond the scope of the issue at hand. But I'll mention this to the team and check what they think. Let's not add now though, thanks!

Reply:
Agreed !!

Reply:
cc @shcheklein @algomaster99 @iAdramelk what do you think about adding `jest` to our pre-commit hook?

Reply:
Thanks for the feedback Alexey. I agree.